### PR TITLE
Diona now have a verb that lets them use magboot-like functionality.

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
title

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!review

File: code/modules/mob/living/carbon/human/diona_gestalt.dm [hunk 1]
```diff
@@ -354,6 +354,44 @@
...
+
+	if(isspaceturf(get_turf(src)))
+		to_chat(src, SPAN_WARNING("You can't do that here!"))
+		return
+
+	last_special = world.time + 2 SECONDS
+
+	if(HAS_TRAIT(src, TRAIT_SHOE_GRIP))
+		visible_message(SPAN_NOTICE("[src] retracts the roots at their feet into their body."), SPAN_NOTICE("You retract your roots."))
+		REMOVE_TRAIT(src, TRAIT_SHOE_GRIP)
```
Comment (alsoandanswer) [inline code/modules/mob/living/carbon/human/diona_gestalt.dm:388]:
```suggestion
		REMOVE_TRAIT(src, TRAIT_SHOE_GRIP, INNATE_TRAIT)
```

File: code/modules/mob/living/carbon/human/diona_gestalt.dm [hunk 2]
```diff
@@ -354,6 +354,44 @@
...
+		playsound(src, 'sound/species/diona/gestalt_split.ogg', 20)
+	else
+		visible_message(SPAN_NOTICE("[src] extends some roots at their feet."), SPAN_NOTICE("You extend your roots to keep yourself upright."))
+		ADD_TRAIT(src, TRAIT_SHOE_GRIP)
```
Comment (alsoandanswer) [inline code/modules/mob/living/carbon/human/diona_gestalt.dm:392]:
```suggestion
		ADD_TRAIT(src, TRAIT_SHOE_GRIP, INNATE_TRAIT)
```

File: code/modules/mob/living/carbon/human/diona_gestalt.dm [hunk 3]
```diff
@@ -354,6 +354,44 @@
 	visible_message("<span class='warning'>\The [src] quivers slightly, then splits apart with a wet slithering noise.</span>")
 	qdel(src)
 
+/mob/living/carbon/human/proc/root_to_ground()
```
Comment (FluffyGhoster) [inline code/modules/mob/living/carbon/human/diona_gestalt.dm:357]: DMDoc this (the name doesn't make readily apparent what this does)

File: code/modules/mob/living/carbon/human/diona_gestalt.dm [hunk 4]
```diff
@@ -354,6 +354,44 @@
...
+	if(isspaceturf(get_turf(src)))
+		to_chat(src, SPAN_WARNING("You can't do that here!"))
+		return
+
+	last_special = world.time + 2 SECONDS
+
+	if(HAS_TRAIT(src, TRAIT_SHOE_GRIP))
+		visible_message(SPAN_NOTICE("[src] retracts the roots at their feet into their body."), SPAN_NOTICE("You retract your roots."))
+		REMOVE_TRAIT(src, TRAIT_SHOE_GRIP, TRAIT_SOURCE_SPECIES_VERB)
+		playsound(src, 'sound/species/diona/gestalt_split.ogg', 20)
```
Comment (FluffyGhoster) [inline code/modules/mob/living/carbon/human/diona_gestalt.dm:389]: Set an appropriate range using one of the defines

File: code/modules/mob/living/carbon/human/diona_gestalt.dm [hunk 5]
```diff
@@ -354,6 +354,44 @@
...
+	else
+		visible_message(SPAN_NOTICE("[src] extends some roots at their feet."), SPAN_NOTICE("You extend your roots to keep yourself upright."))
+		ADD_TRAIT(src, TRAIT_SHOE_GRIP, TRAIT_SOURCE_SPECIES_VERB)
+		playsound(src, 'sound/species/diona/gestalt_grow.ogg', 30)
```
Comment (FluffyGhoster) [inline code/modules/mob/living/carbon/human/diona_gestalt.dm:393]: Set an appropriate range using one of the defines

**Review by Alberyk — APPROVED** [review]

File: code/modules/mob/living/carbon/human/diona_gestalt.dm [hunk 6]
```diff
@@ -359,6 +359,47 @@
...
+		to_chat(src, SPAN_WARNING("You've used an ability too recently! Wait a bit."))
+		return
+
+	if(!has_organ(BP_R_FOOT) && !has_organ(BP_L_FOOT))
+		to_chat(src, SPAN_WARNING("You have no supporting limbs to extend your roots from!"))
+		return
+
+	if(isspaceturf(get_turf(src)))
+		to_chat(src, SPAN_WARNING("You can't do that here!"))
+		return
```
Comment (Arrow768) [inline code/modules/mob/living/carbon/human/diona_gestalt.dm:390]: I currently only have my phone available, so searching through the code is rather complicated.

It seems that all these checks are only performed once.

However if these checks are required to enable the function, they should also be performed regularly while using it.

There should also be some cost to using the "magboot" function, either in the form of decreased movement speed or increased hunger (or something else?)
Is that already handled somewhere else (by the trait?)?

**Review by Arrow768 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**